[Bug] `mr close` panics with index out of range [0] with length 0

`glab mr close` panics when merge request ID is not provided.

**Expected Behaviour**

If the merge request id is not provided, `glab` should find the open merge request for the current checked out branch and close it

**Actual Behaviour**
```sh
→ glab mr close
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/profclems/glab/commands/mr/close.NewCmdClose.func1(0xc0002e8580, 0x14a4e48, 0x0, 0x0, 0x0, 0x0)
        github.com/profclems/glab/commands/mr/close/mr_close.go:43 +0x605
github.com/spf13/cobra.(*Command).execute(0xc0002e8580, 0x14a4e48, 0x0, 0x0, 0xc0002e8580, 0x14a4e48)
        github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000405080, 0xc000102160, 0x2, 0x2)
        github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
main.main()
        github.com/profclems/glab/cmd/glab/main.go:114 +0x736
```

It works when merge request ID is provided
```sh
→ glab mr close 359
- Closing Merge request...✔ Merge request !359
!359 fix: error 500 when creating purchase order' (fix-err-500-purchase-order)
 https://gitlab.com/ssisgh-projects/simeds/back-end/-/merge_requests/359
```

--- a/commands/mr/close/mr_close.go
+++ b/commands/mr/close/mr_close.go
@@ -2,7 +2,6 @@ package close
 
 import (
 	"fmt"
-	"strings"
 
 	"github.com/profclems/glab/commands/cmdutils"
 	"github.com/profclems/glab/commands/mr/mrutils"
@@ -18,7 +17,6 @@ func NewCmdClose(f *cmdutils.Factory) *cobra.Command {
 		Use:   "close [<id> | <branch>]",
 		Short: `Close merge requests`,
 		Long:  ``,
-		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			var err error
 			apiClient, err := f.HttpClient()
@@ -26,30 +24,26 @@ func NewCmdClose(f *cmdutils.Factory) *cobra.Command {
 				return err
 			}
 
-			mr, repo, err := mrutils.MRFromArgs(f, args)
+			mrs, repo, err := mrutils.MRsFromArgs(f, args)
 			if err != nil {
 				return err
 			}
 
-			if err = mrutils.MRCheckErrors(mr, mrutils.MRCheckErrOptions{
-				Closed: true,
-				Merged: true,
-			}); err != nil {
-				return err
-			}
-
-			mergeID := args[0]
-
 			l := &gitlab.UpdateMergeRequestOptions{}
 			l.StateEvent = gitlab.String("close")
-			arrIds := strings.Split(strings.Trim(mergeID, "[] "), ",")
-			for _, i2 := range arrIds {
+			for _, mr := range mrs {
+				if err = mrutils.MRCheckErrors(mr, mrutils.MRCheckErrOptions{
+					Closed: true,
+					Merged: true,
+				}); err != nil {
+					return err
+				}
 				fmt.Fprintf(f.IO.StdOut, "- Closing Merge request...")
-				mr, err := api.UpdateMR(apiClient, repo.FullName(), utils.StringToInt(i2), l)
+				_, err := api.UpdateMR(apiClient, repo.FullName(), mr.IID, l)
 				if err != nil {
 					return err
 				}
-				fmt.Fprintf(f.IO.StdOut, "%s Merge request !%s\n", utils.RedCheck(), i2)
+				fmt.Fprintf(f.IO.StdOut, "%s Merge request !%d\n", utils.RedCheck(), mr.IID)
 				fmt.Fprintln(f.IO.StdOut, mrutils.DisplayMR(mr))
 			}
 

--- a/commands/mr/close/mr_close.go
+++ b/commands/mr/close/mr_close.go
@@ -17,6 +17,7 @@ func NewCmdClose(f *cmdutils.Factory) *cobra.Command {
 		Use:   "close [<id> | <branch>]",
 		Short: `Close merge requests`,
 		Long:  ``,
+		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			var err error
 			apiClient, err := f.HttpClient()

--- a/commands/mr/delete/mr_delete.go
+++ b/commands/mr/delete/mr_delete.go
@@ -16,6 +16,7 @@ func NewCmdDelete(f *cmdutils.Factory) *cobra.Command {
 		Use:     "delete [<id> | <branch>]",
 		Short:   `Delete merge requests`,
 		Long:    ``,
+		Args:    cobra.MaximumNArgs(1),
 		Aliases: []string{"del"},
 		Example: "$ glab delete 123",
 		RunE: func(cmd *cobra.Command, args []string) error {

--- a/commands/mr/delete/mr_delete.go
+++ b/commands/mr/delete/mr_delete.go
@@ -17,7 +17,6 @@ func NewCmdDelete(f *cmdutils.Factory) *cobra.Command {
 		Short:   `Delete merge requests`,
 		Long:    ``,
 		Aliases: []string{"del"},
-		Args:    cobra.MaximumNArgs(1),
 		Example: "$ glab delete 123",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			apiClient, err := f.HttpClient()
@@ -25,18 +24,18 @@ func NewCmdDelete(f *cmdutils.Factory) *cobra.Command {
 				return err
 			}
 
-			mr, repo, err := mrutils.MRFromArgs(f, args)
+			mrs, repo, err := mrutils.MRsFromArgs(f, args)
 			if err != nil {
 				return err
 			}
 
-			fmt.Fprintf(f.IO.StdOut, "- Deleting Merge Request !%d\n", mr.IID)
-
-			if err = api.DeleteMR(apiClient, repo.FullName(), mr.IID); err != nil {
-				return err
+			for _, mr := range mrs {
+				fmt.Fprintf(f.IO.StdOut, "- Deleting Merge Request !%d\n", mr.IID)
+				if err = api.DeleteMR(apiClient, repo.FullName(), mr.IID); err != nil {
+					return err
+				}
+				fmt.Fprintf(f.IO.StdOut, "%s Merge request !%d deleted\n", utils.RedCheck(), mr.IID)
 			}
-
-			fmt.Fprintf(f.IO.StdOut, "%s Merge request !%d deleted\n", utils.RedCheck(), mr.IID)
 
 			return nil
 		},

--- a/commands/mr/mrutils/mrutils.go
+++ b/commands/mr/mrutils/mrutils.go
@@ -1,6 +1,7 @@
 package mrutils
 
 import (
+	"context"
 	"fmt"
 	"strconv"
 
@@ -10,6 +11,7 @@ import (
 	"github.com/profclems/glab/pkg/api"
 	"github.com/profclems/glab/pkg/tableprinter"
 	"github.com/xanzy/go-gitlab"
+	"golang.org/x/sync/errgroup"
 )
 
 type MRCheckErrOptions struct {
@@ -149,6 +151,54 @@ func MRFromArgs(f *cmdutils.Factory, args []string) (*gitlab.MergeRequest, glrep
 	}
 
 	return mr, baseRepo, nil
+}
+
+func MRsFromArgs(f *cmdutils.Factory, args []string) ([]*gitlab.MergeRequest, glrepo.Interface, error) {
+	if len(args) <= 1 {
+		mrID, baseRepo, err := MRFromArgs(f, args)
+		if err != nil {
+			return nil, nil, err
+		}
+		return []*gitlab.MergeRequest{mrID}, baseRepo, err
+	}
+
+	apiClient, err := f.HttpClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	baseRepo, err := f.BaseRepo()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	errGroup, _ := errgroup.WithContext(context.Background())
+	mrs := make([]*gitlab.MergeRequest, len(args))
+	for i, arg := range args {
+		i, arg := i, arg
+		errGroup.Go(func() error {
+			mrID, err := strconv.Atoi(arg)
+			if err != nil {
+				return err
+			}
+			if mrID == 0 {
+				return fmt.Errorf("invalid merge request ID provided")
+			}
+			// fetching multiple MRs does not return many major params in the payload
+			// so we fetch again using the single mr endpoint
+			mr, err := api.GetMR(apiClient, baseRepo.FullName(), mrID, &gitlab.GetMergeRequestsOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get merge request %d: %w", mrID, err)
+			}
+			mrs[i] = mr
+			return nil
+		})
+	}
+	if err := errGroup.Wait(); err != nil {
+		return nil, nil, err
+	}
+	return mrs, baseRepo, nil
+
 }
 
 func GetOpenMRForBranch(apiClient *gitlab.Client, baseRepo glrepo.Interface, currentBranch string) (*gitlab.MergeRequest, error) {

--- a/commands/mr/mrutils/mrutils.go
+++ b/commands/mr/mrutils/mrutils.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"strconv"
+	"strings"
 
 	"github.com/profclems/glab/commands/cmdutils"
 	"github.com/profclems/glab/internal/glrepo"
@@ -155,11 +156,18 @@ func MRFromArgs(f *cmdutils.Factory, args []string) (*gitlab.MergeRequest, glrep
 
 func MRsFromArgs(f *cmdutils.Factory, args []string) ([]*gitlab.MergeRequest, glrepo.Interface, error) {
 	if len(args) <= 1 {
-		mrID, baseRepo, err := MRFromArgs(f, args)
-		if err != nil {
-			return nil, nil, err
+		var arrIDs []string
+		if len(args) == 1 {
+			arrIDs = strings.Split(args[0], ",")
 		}
-		return []*gitlab.MergeRequest{mrID}, baseRepo, err
+		if len(arrIDs) <= 1 {
+			mr, baseRepo, err := MRFromArgs(f, args)
+			if err != nil {
+				return nil, nil, err
+			}
+			return []*gitlab.MergeRequest{mr}, baseRepo, err
+		}
+		args = arrIDs
 	}
 
 	apiClient, err := f.HttpClient()

--- a/commands/mr/reopen/mr_reopen.go
+++ b/commands/mr/reopen/mr_reopen.go
@@ -18,37 +18,36 @@ func NewCmdReopen(f *cmdutils.Factory) *cobra.Command {
 		Short:   `Reopen merge requests`,
 		Long:    ``,
 		Aliases: []string{"open"},
-		Args:    cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			apiClient, err := f.HttpClient()
 			if err != nil {
 				return err
 			}
 
-			mr, repo, err := mrutils.MRFromArgs(f, args)
+			mrs, repo, err := mrutils.MRsFromArgs(f, args)
 			if err != nil {
-				return err
-			}
-
-			if err = mrutils.MRCheckErrors(mr, mrutils.MRCheckErrOptions{
-				Opened: true,
-				Merged: true,
-			}); err != nil {
 				return err
 			}
 
 			l := &gitlab.UpdateMergeRequestOptions{}
 			l.StateEvent = gitlab.String("reopen")
+			for _, mr := range mrs {
+				if err = mrutils.MRCheckErrors(mr, mrutils.MRCheckErrOptions{
+					Opened: true,
+					Merged: true,
+				}); err != nil {
+					return err
+				}
 
-			fmt.Fprintf(f.IO.StdOut, "- Reopening Merge request !%d...\n", mr.IID)
+				fmt.Fprintf(f.IO.StdOut, "- Reopening Merge request !%d...\n", mr.IID)
+				mr, err = api.UpdateMR(apiClient, repo.FullName(), mr.IID, l)
+				if err != nil {
+					return err
+				}
 
-			mr, err = api.UpdateMR(apiClient, repo.FullName(), mr.IID, l)
-			if err != nil {
-				return err
+				fmt.Fprintf(f.IO.StdOut, "%s Merge request !%d reopened\n", utils.GreenCheck(), mr.IID)
+				fmt.Fprintln(f.IO.StdOut, mrutils.DisplayMR(mr))
 			}
-
-			fmt.Fprintf(f.IO.StdOut, "%s Merge request !%d reopened\n", utils.GreenCheck(), mr.IID)
-			fmt.Fprintln(f.IO.StdOut, mrutils.DisplayMR(mr))
 
 			return nil
 		},

--- a/commands/mr/reopen/mr_reopen.go
+++ b/commands/mr/reopen/mr_reopen.go
@@ -17,6 +17,7 @@ func NewCmdReopen(f *cmdutils.Factory) *cobra.Command {
 		Use:     "reopen [<id> | <branch>]",
 		Short:   `Reopen merge requests`,
 		Long:    ``,
+		Args:    cobra.MaximumNArgs(1),
 		Aliases: []string{"open"},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			apiClient, err := f.HttpClient()

--- a/go.mod
+++ b/go.mod
@@ -35,6 +35,7 @@ require (
 	github.com/yuin/goldmark v1.2.1 // indirect
 	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
 	golang.org/x/net v0.0.0-20200822124328-c89045814202 // indirect
+	golang.org/x/sync v0.0.0-20190423024810-112230192c58
 	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c
 )

